test/test_library_loading.cpp

```diff
@@ -0,0 +1,171 @@
+#include "dlibxx.hxx"
+
+#include <gmock/gmock.h>
+#include "mocks/dlfcn_mock.h"
+
+#include <tuple>
+
+using namespace testing;
+
+class TestLoadLibrary : public Test
+{
+public:
+	TestLoadLibrary()
+		: Test{}
+	{
+		ON_CALL(dlfcn, dlopen(_,_)).WillByDefault(Return(reinterpret_cast<void*>(1)));
+		ON_CALL(dlfcn, dlclose(_)).WillByDefault(Return(0));
+	}
+
+protected:
+	NiceMock<DlfcnMock> dlfcn;
+	dlibxx::handle lib;
+};
+
+TEST_F(TestLoadLibrary, Loaded_WhenLibraryCannotBeFound_WillReturnFalse)
+{
+	auto const lib_name = std::string{ "test"};
+	EXPECT_CALL(dlfcn, dlopen(StrEq(lib_name),_)).WillRepeatedly(Return(nullptr));
+
+	lib.load(lib_name);
+
+	EXPECT_FALSE(lib.loaded());
+}
+
+TEST_F(TestLoadLibrary, Loaded_WhenLibraryIsFound_WillReturnTrue)
+{
+	auto const lib_name = std::string{ "test"};
+	EXPECT_CALL(dlfcn, dlopen(StrEq(lib_name),_)).WillRepeatedly(Return(reinterpret_cast<void*>(1)));
+
+	lib.load(lib_name);
+
+	EXPECT_TRUE(lib.loaded());
+}
+
+TEST_F(TestLoadLibrary, OperatorBool_WhenLibraryCannotBeFound_WillReturnFalse)
+{
+	auto const lib_name = std::string{ "test"};
+	EXPECT_CALL(dlfcn, dlopen(StrEq(lib_name),_)).WillRepeatedly(Return(nullptr));
+
+	lib.load(lib_name);
+
+	EXPECT_FALSE(lib);
+}
+
+TEST_F(TestLoadLibrary, OperatorBool_WhenLibraryIsFound_WillReturnTrue)
+{
+	auto const lib_name = std::string{ "test"};
+	EXPECT_CALL(dlfcn, dlopen(StrEq(lib_name),_)).WillRepeatedly(Return(reinterpret_cast<void*>(1)));
+
+	lib.load(lib_name);
+
+	EXPECT_TRUE(lib);
+}
+
+class TestDlopenFlags : public TestLoadLibrary,
+                        public WithParamInterface<std::tuple<dlibxx::resolve, dlibxx::options>>
+{
+};
+
+TEST_P(TestDlopenFlags, Load_WillCorrectlyPassFlagsToDlopen)
+{
+	auto const resolve_policy  = std::get<0>(GetParam());
```

Comment:
Can you put the type here? It's not obvious from context what it is. Same for other uses of `auto` below.

Reply:
Whoops! I had tried to write `auto const resolve_policy = std::get<dlibxx::resolve>(GetParam())` but it didn't seem to work... Similarly for the next line. I will change this.